server/conn.go

```diff
@@ -160,6 +160,18 @@ func (cc *clientConn) Close() error {
 	return nil
 }
 
+func (cc *clientConn) CloseWithoutLock() error {
```

Comment:
you can use this function in clientConn.Close(), never write the same code in two places.